interface/backend/api/views.py

```diff
@@ -76,15 +76,28 @@ def filename_to_dict(name, location):
         }
         return d
 
+    @staticmethod
+    def is_hidden(location):
+        """
+        Check whether a file or a directory is hidden.
+        """
+        return os.path.basename(location).startswith('.')
+
     def walk(self, location, dir_name='/'):
         """
         Recursively walkthrough directories and files
         """
-        folders, files = self.fss.listdir(location)
-        tree = {'name': dir_name, 'children': []}
-        tree['files'] = [self.filename_to_dict(filename, location) for filename in sorted(files)]
-        tree['type'] = 'folder'
-        tree['children'] = [self.walk(os.path.join(location, dir), dir) for dir in folders]
+        tree = None
+        if not self.is_hidden(location):
+            folders, files = self.fss.listdir(location)
+            tree = {'name': dir_name, 'children': []}
```

Comment:
Why not just define ``tree = { .. }`` all in one go, like:

```
tree = {
    'name': dir_name
    'type': 'folder',
    'children': […],
    'files': […],
}
```

Assigning and then re-assigning makes me think (incorrectly) that we are *re-using* values which is misleading

Reply:
`tree` is now defined in one go, additionally, I've assumed that initial location shall not be hidden.